_projects/2023/100517886/100517886.Rmd

```diff
@@ -0,0 +1,454 @@
+author: Sara Cristina Herranz Amado
+date: "`r Sys.Date()`"
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+---
+
+git add
+git commit 
+git push
+en algun lugar tengo que añadir _project. 
+es lo que estuvimos haciendo el otro día
+
+
+Distill is a publication format for scientific and technical writing, native to the web.
+
+Learn more about using Distill at <https://rstudio.github.io/distill>.
+
+```{r}
+library(readr)
+library(readxl)
+library(tidyverse)
+library(ggplot2)
+library(ggstream)
+library(dplyr)
+library(ggtext)
+library(ggrepel)
+library(showtext)
+library(sysfonts)
+library(diagram)
+library(grid)
+library(plotrix)
+```
+
+## **CLEANING THE DATA**
+
+Uploading the data base, deleting unnecessary columns and NAs from the main variable (oilprod_kbd):
+
+```{r}
+data <-  read_xlsx(path = "Panel format.xlsx", sheet = 1, range = "A1:M7212")
+
+data <- data |>
+  select(-c(OPEC:CIS))
+
+panel <- data |> 
+  filter(!is.na(oilprod_kbd)) 
+  
+```
+
+First, we start by organizing the countries. We created a vector for the countries that needed to appear explicitly in our graph and then created a new database where all these countries appear. After that, we filtered the original database, but this time we selected the unwanted countries and summed them by region. That way, they would receive the name of the region.
+
+```{r}
+countries <- c("Australia", "China", "Indonesia", "Norway", "Russian Federation", "USSR", "United Kingdom", "Brazil", "Venezuela", "Mexico", "Canada", "US", "Saudi Arabia", "Iran", "Kuwait", "Iraq", "Libya", "United Arab Emirates", "Nigeria", "Other Africa", "Other Asia Pacific", "Other CIS", "Other Caribbean", "Other Eastern Africa", "Other Europe", "Other Middle Africa", "Other Middle East", "Other Northern Africa", "Other S. & Cent. America", "Other South America", "Other Western Africa")
+
+# Filter data by country of interest
+panel_a <-  panel |>
+  group_by(Region, Country) |> 
+  filter(Country %in% countries) |> 
+  mutate(Region = if_else(Country == "Australia", "Oceania", Region)) |> 
+  select(-c(pop, ISO3166_alpha3, ISO3166_numeric, SubRegion, oilprod_mt)) 
+
+  
+# Filter data by countries that are not of our interest and summing them 
+panel_b <- panel |> 
+  filter(!Country %in% countries) |> 
+  group_by(Year, Region) |> 
+  summarise(oilprod_kbd = sum(oilprod_kbd), .groups = 'drop') |> 
+  mutate(Country = Region) |> 
+  select(Country, everything()) |>
+  filter(!is.na(Region)) |> 
+  arrange(Country) 
+```
+
+We combined both data bases in order to work with them. To do so, we changed oilprod_kbd, the main variable, as a character variable so it would not change the data, since we were having some trouble. After that, we create the new data base, panel_c, and change oilprod_kbd into a numeric variable again.
+
+```{r include=FALSE}
+  # defining oilprod_kbd as character in both data bases
+as.character(panel_a$oilprod_kbd)
+as.character(panel_b$oilprod_kbd)
```

Comment:
Please, remove this. There's a lot of output here that is not useful for the reader.